CMakeLists.txt

```diff
@@ -294,15 +295,14 @@ if((NOT WITH_API_ONLY)
         include_directories(
           ${PROJECT_SOURCE_DIR}/third_party/nlohmann-json/single_include)
       else()
-        message(
-          FATAL_ERROR
-            "\nnlohmann_json package was not found. Please either provide it manually or clone with submodules. "
-            "To initialize, fetch and checkout any nested submodules, you can use the following command:\n"
-            "git submodule update --init --recursive")
+        set(nlohmann_json_clone TRUE)
+        include(cmake/nlohmann-json.cmake)
+        message("\nnlohmann_json package was not found. Cloning from github")
       endif()
     else()
-      # Let's fail with default find_package error
-      find_package(nlohmann_json REQUIRED)
+      set(nlohmann_json_clone TRUE)
+      include(cmake/nlohmann-json.cmake)
+      message("\nnlohmann_json package was not found. Cloning from github")
```

Comment:
Just wondering if we can bit simplify from L281 - L308 as ( not tested ) :

```
set(nlohmann_json_clone FALSE)
if (nlohmann_json_FOUND)
  message("Using external nlohmann::json")
elseif( EXISTS ${PROJECT_SOURCE_DIR}/.git AND EXISTS ${PROJECT_SOURCE_DIR}/third_party/nlohmann-json/CMakeLists.txt)
       message("Trying to use local nlohmann::json from submodule")
	...
        include_directories(
          ${PROJECT_SOURCE_DIR}/third_party/nlohmann-json/single_include)
else()
      set(nlohmann_json_clone TRUE)
      include(cmake/nlohmann-json.cmake)
      message("\nnlohmann_json package was not found. Cloning from github")
endif()
```

Reply:
thanks. done